src/app-layout/interfaces.ts

```diff
@@ -337,14 +337,19 @@ export namespace AppLayoutProps {
 }
 
 export type AppLayoutPropsWithDefaults = SomeRequired<
```

Comment:
`AppLayoutPropsWithDefaults` did not really kept up to its name, it started doing more things than the name says.

Now there are separate `AppLayoutPropsWithDefaults` (only some defaults) and `AppLayoutInternalProps` with additional prop extensions for internal use